packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -638,7 +638,7 @@ class DevFS {
       bundle.entries.forEach((String archivePath, AssetBundleEntry entry) {
         // If the content is backed by a real file, isModified will file stat and return true if
         // it was modified since the last time this was called.
-        if (!entry.content.isModified || bundleFirstUpload) {
```

Comment:
Here is what I don't like. Callers of this API need to be sure to read `entry.isModified` and not `entry.content.isModified`. The latter really shouldn't be exposed, but it has to be because we have logic that depends on the runtime type of `entry.content`.